File: MiniDOMAsteriskLib/CallManagers/Events/Newchannel.cs

namespace minidom.CallManagers.Events
{
    public class Newchannel : AsteriskEvent
    {
        public Newchannel() : base("Newchannel")
        {
        }

        public Newchannel(string[] rows) : base(rows)
        {
        }

        public string State
        {
            get
            {
                return GetAttribute("State");
            }
        }

        public string Channel
        {
            get
            {
                return GetAttribute("Channel");
            }
        }

        public string CallerID
        {
            get
            {
                return GetAttribute("CallerID");
            }
        }

        public string CallerIDNum
        {
            get
            {
                return GetAttribute("CallerIDNum");
            }
        }

        public string CallerIDName
        {
            get
            {
                return GetAttribute("CallerIDName");
            }
        }

        public string UniqueID
        {
            get
            {
                return GetAttribute("UniqueID");
            }
        }
    }
}

File: MiniDOMAsteriskLib/CallManagers/Events/Newchannel.vb
Imports System.Net.Sockets
Imports System.Threading
Imports System.Net

Namespace CallManagers.Events

    
    Public Class Newchannel
        Inherits AsteriskEvent

        Public Sub New()
            MyBase.New("Newchannel")
        End Sub

        Public Sub New(ByVal rows() As String)
            MyBase.New(rows)
        End Sub

        Public ReadOnly Property State As String
            Get
                Return Me.GetAttribute("State")
            End Get
        End Property

        Public ReadOnly Property Channel As String
            Get
                Return Me.GetAttribute("Channel")
            End Get
        End Property

        Public ReadOnly Property CallerID As String
            Get
                Return Me.GetAttribute("CallerID")
            End Get
        End Property

        Public ReadOnly Property CallerIDNum As String
            Get
                Return Me.GetAttribute("CallerIDNum")
            End Get
        End Property

        Public ReadOnly Property CallerIDName As String
            Get
                Return Me.GetAttribute("CallerIDName")
            End Get
        End Property

        Public ReadOnly Property UniqueID As String
            Get
                Return Me.GetAttribute("UniqueID")
            End Get
        End Property
         
    End Class

End Namespace